applications/main/infrared/scenes/infrared_scene_rpc.c

```diff
@@ -124,6 +124,53 @@ bool infrared_scene_rpc_on_event(void* context, SceneManagerEvent event) {
 
             rpc_system_app_confirm(infrared->rpc_ctx, result);
 
+        } else if(
+            event.event == InfraredCustomEventTypeRpcButtonPressReleaseName ||
+            event.event == InfraredCustomEventTypeRpcButtonPressReleaseIndex) {
+            bool result = false;
+
+            // Send the signal once and stop
+            if(rpc_state == InfraredRpcStateLoaded) {
+                if(event.event == InfraredCustomEventTypeRpcButtonPressReleaseName) {
+                    const char* button_name = furi_string_get_cstr(infrared->button_name);
+                    size_t index;
+                    const bool index_found =
+                        infrared_remote_get_signal_index(infrared->remote, button_name, &index);
+                    app_state->current_button_index = index_found ? (signed)index :
+                                                                    InfraredButtonIndexNone;
+                    FURI_LOG_D(TAG, "Sending signal with name \"%s\"", button_name);
+                } else {
+                    FURI_LOG_D(
+                        TAG, "Sending signal with index \"%ld\"", app_state->current_button_index);
+                }
+                if(infrared->app_state.current_button_index != InfraredButtonIndexNone) {
+                    InfraredErrorCode error =
+                        infrared_tx_start_button_index(infrared, app_state->current_button_index);
```

Comment:
There is no guarantee that the signal will be sent only once, as there is a worker being asynchronously spun up only to be stopped right away.
Try using `infrared_signal_transmit` instead (see how it is done in `infrared_brute_force.c`)